src/others/helpers/aid-request-helpers.ts

```diff
@@ -0,0 +1,93 @@
+import { groupBy, map, filter, eq, flatten } from "lodash";
+import { Location, Supply, AidRequest, ID } from "../contexts/api";
+
+import { ListItem } from "../components/CollapsibleListItem";
+
+export const translateToLocation =
+  (cities: Location[]) =>
+  (city_id: number): Location => {
+    return cities[city_id - 1];
+  }; // TODO: remove once map consumes DictionaryContext properly
+
+export const translateToSupply =
+  (supplies: Supply[]) =>
+  (category_id: ID): Supply => {
+    const supply = supplies.find((suply) => String(suply.id) === String(category_id));
+    if (!supply) throw new Error(`Supply category: ${category_id} is not defined`);
+
+    return supply;
+  }; // TODO: remove once map consumes DictionaryContext properly
+
+export const sortDates = (a: string, b: string) => {
+  return new Date(a).getTime() - new Date(b).getTime();
+};
+
+export const filterByCategoryIds = (aidRequests: AidRequest[], categoryIds: (string | "*")[]): AidRequest[] => {
+  if (categoryIds.includes("*")) return aidRequests;
+  return flatten(categoryIds.map((categoryId) => filter(aidRequests, (req) => eq(categoryId, req.category_id))));
```

Comment:
The complexity of this line seems to be O(num_categories * num_of_requests)
Why cannot we just build a set of selected categories and then filter the data in one run? Thus the complexity is O(num_categories + num_of_requests).
(+ instead of *)

Reply:
I hope that's what you're thinking of: 

```ts
export const filterByCategoryIds = (aidRequests: AidRequest[], categoryIds: (string | "*")[]): AidRequest[] => {
  if (categoryIds.includes("*")) return aidRequests;
  return aidRequests.filter((aidRequest) => categoryIds.includes(aidRequest.category_id));
};
```

Reply:
much closer, but categoryIds is still a list, so i suspect that ".includes" takes linear time.
Is it possible to convert it to a set first so that ".includes" could be checked in constant time?